src/common.c

```diff
@@ -64,7 +65,20 @@ int fi_poll_fd(int fd, int timeout)
 
 	fds.fd = fd;
 	fds.events = POLLIN;
-	ret = poll(&fds, 1, timeout);
+
+	// to set the signal mask
+        sigset_t orgmask, sigmask;
+
+         /* Block SIGUSR1, SIGUSR2, and all realtime signals from SIGRTMIN to SIGRTMAX */
+        sigemptyset(&sigmask);
+        sigaddset(&sigmask, SIGUSR1);
+        sigaddset(&sigmask, SIGUSR2);
+	for (int i = SIGRTMIN; i <= SIGRTMAX; i++)
+	        sigaddset(&sigmask, i);
+        pthread_sigmask(SIG_SETMASK, &sigmask, &orgmask);
+        ret = poll(&fds, 1, timeout);
+        pthread_sigmask(SIG_SETMASK, &orgmask, NULL);
```

Comment:
I have looked into the man pages, and I still think the code is safe but I am not 100% sure about it. My understanding is that the race may occur when the signal handler writes into a memory location that the libfabric polling routine might rely on for control flow. I don't think the user will do such thing in the signal handler since the user is not supposed to see internal data of libfabric. But I am not sure if this can occur indirectly by calling an libfabric API routine. Feel free to close the PR; I will keep patching my fork when needed.